drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-businesscentral-client/src/main/java/org/drools/workbench/screens/scenariosimulation/businesscentral/client/editor/ScenarioSimulationEditorBusinessCentralWrapper.java

```diff
@@ -0,0 +1,312 @@
+        placeManager.registerOnOpenCallback(new DefaultPlaceRequest(TestToolsPresenter.IDENTIFIER), scenarioSimulationEditorPresenter.getPopulateTestToolsCommand());
+    }
+
+    protected void unRegisterTestToolsCallback() {
+        placeManager.getOnOpenCallbacks(new DefaultPlaceRequest(TestToolsPresenter.IDENTIFIER)).remove(scenarioSimulationEditorPresenter.getPopulateTestToolsCommand());
+    }
+
+    /**
+     * If you want to customize the menu override this method.
+     */
+    @Override
+    protected Promise<Void> makeMenuBar() {
+        scenarioSimulationEditorPresenter.makeMenuBar(fileMenuBuilder);
+        return super.makeMenuBar();
+    }
+
+    @Override
+    protected Supplier<ScenarioSimulationModel> getContentSupplier() {
+        return scenarioSimulationEditorPresenter.getContentSupplier();
+    }
+
+    @Override
+    protected void save(final String commitMessage) {
+        final ScenarioSimulationModel model = scenarioSimulationEditorPresenter.getModel();
+        RemoteCallback<Path> saveSuccessCallback = getSaveSuccessCallback(scenarioSimulationEditorPresenter.getJsonModel(model).hashCode());
+        service.call(saveSuccessCallback,
+                     new HasBusyIndicatorDefaultErrorCallback(scenarioSimulationEditorPresenter.getView())).save(versionRecordManager.getCurrentPath(),
+                                                                                                                 model,
+                                                                                                                 metadata,
+                                                                                                                 commitMessage);
+    }
+
+    @Override
+    protected void addCommonActions(final FileMenuBuilder fileMenuBuilder) {
+        scenarioSimulationEditorPresenter.addCommonActions(fileMenuBuilder, versionRecordManager.buildMenu(), alertsButtonMenuItemBuilder.build());
+    }
+
+    @Override
+    protected void loadContent() {
+        scenarioSimulationEditorPresenter.loadContent();
+        service.call(getModelSuccessCallback(),
+                     getNoSuchFileExceptionErrorCallback()).loadContent(versionRecordManager.getCurrentPath());
+    }
+
+    @Override
+    protected Caller<? extends SupportsDelete> getDeleteServiceCaller() {
+        return service;
+    }
+
+    @Override
+    protected Caller<? extends SupportsRename> getRenameServiceCaller() {
+        return service;
+    }
+
+    @Override
+    protected Caller<? extends SupportsCopy> getCopyServiceCaller() {
+        return service;
+    }
+
+    @Override
+    protected Caller<? extends SupportsSaveAndRename<ScenarioSimulationModel, Metadata>> getSaveAndRenameServiceCaller() {
+        return service;
+    }
+
+    @Override
+    protected String getEditorIdentifier() {
+        return IDENTIFIER;
+    }
+
+    protected void getModelSuccessCallbackMethod(ScenarioSimulationModelContent content) {
```

Comment:
I would like to have all these logic in the presenter instead of the wrapper but I agree it could be complex to do now so if you agree please create a ticket to cover this "refactor"

Reply:
@danielezonca 
OK, created https://issues.jboss.org/browse/DROOLS-4153